Remove bit fields and use fixed width integers instead.

madmiraal opened the pull request:
As identified by [lgtm](https://lgtm.com/projects/g/godotengine/godot/alerts/?mode=list&lang=&id=cpp%2Fambiguously-signed-bit-field), the signedness of `int` bit fields is implementation specific. Therefore, they need to be explicitly signed or unsigned. Since -1 is used to identify an unconfigured `width_cache` or `wrap_amount_cache`, this PR makes them explicitly signed.

madmiraal committed:
Remove bit fields and use fixed width integers instead.

aaronfranke commented:
Wouldn't it be better to use `int32_t` or similar?

Paulb23 reviewed (commented):
Yeah probably best to use `int32_t` here.

madmiraal commented:
Changed to use `int32_t` instead, and removed the bit fields on the `bool`s, because bit fields are slow, and the goal doesn't seem to be to save space. Also, rebased following #40973.

aaronfranke commented:
What do you mean bit fields? Also, doesn't getting rid of `: 24` etc change the behavior?

madmiraal commented:
> What do you mean bit fields? Also, doesn't getting rid of : 24 etc change the behavior?

The `: 24` fixes the size of the variable to the specified number of bits; in this case 24. Using `int32_t` has the same effect as using `: 32`; although it's not done using bit fields. It allows data to be tightly packed to save space. See [cppreference.com](https://en.cppreference.com/w/cpp/language/bit_field) for more information on bit fields. Removing it doesn't change the behaviour, just the amount of space being used (and it eliminates the additional time taken to access bit field data).

Paulb23 reviewed (approved):
LGTM.

akien-mga commented:
Thanks!
